src/replication.c

```diff
@@ -925,6 +940,14 @@ void replconfCommand(client *c) {
              * to the slave. */
             if (server.masterhost && server.master) replicationSendAck();
             return;
+        } else if (!strcasecmp(c->argv[j]->ptr,"rdb-only")) {
+           /* REPLCONF RDB-ONLY is used to identify the client only wants
+            * RDB snapshot without replication buffer. */
+            long rdb_only = 0;
+            if (getLongFromObjectOrReply(c,c->argv[j+1],
+                    &rdb_only,NULL) != C_OK)
+                return;
+            if (rdb_only == 1) c->flags |= CLIENT_REPL_RDBONLY;
```

Comment:
I find it silly we are explicitly checking for a value of 1 here.  We already have single argument replconf commands that just ignore the required second argument. 

I would prefer to have a general mechanism to indicate what the replica wants, so we can have something like "request-type" "rdb-only" or just ignore the optional argument.

Reply:
In theory we could have used
```c
else c->flags &= ~CLIENT_REPL_RDBONLY;
```
It'll make the code less ugly, but I don't think anyone will ever use it to turn off a flag that was previously turned on. 

It might be convenient for some piece of code to always send the replconf upon connecting and just change the argument, but that's silly. 

I don't like `request-type` since I'm not sure it'll be useful and how.. Maybe we'll want to add more types in the future, but it's likely they'll be flags and not an enum. 

So I'm OK with not checking the value, but I guess it's cleaner to do the `else` even if it makes little sense

Reply:
The else would at last clarify the code more. 

Instead of request-type, we could have something more literal like "set-flag" <enum>. "SET-FLAG RDB-ONLY". Anything that is an enum could go there.

Reply:
Thanks @oranagra @madolson i add a commit that handles `replconf rdb-only 0`.
I am not sure we should improve scalability by 'set-flag' or others